source/tests/system/nidaqmx_driver_api_tests.cpp

```diff
@@ -714,16 +725,11 @@ class NiDAQmxDriverApiTests : public Test {
     EXPECT_EQ(::grpc::Status::OK.error_code(), status.error_code());
   }
 
-  void EXPECT_DAQ_ERROR(int32_t expected_error, const std::string& error_message)
+  void EXPECT_DAQ_ERROR(int32_t expected_error, const nidevice_grpc::experimental::client::grpc_driver_error& ex)
   {
-    auto error = json::parse(error_message);
-    EXPECT_EQ(expected_error, error.value("code", 0));
-  }
-
-  void EXPECT_DAQ_ERROR(int32_t expected_error, const ::grpc::Status& status)
-  {
-    EXPECT_EQ(::grpc::StatusCode::UNKNOWN, status.error_code());
-    EXPECT_DAQ_ERROR(expected_error, status.error_message());
+    EXPECT_EQ(::grpc::StatusCode::UNKNOWN, ex.StatusCode());
+    const auto& error = ex.Trailers().find("ni-error")->second;
```

Comment:
I'm not a fan of tests that crash if the SUT doesn't return an error, but I guess these checks are all inside catch blocks, so they would only crash if the SUT returned a non-driver error.

Single-sourcing the error macro would make it less tedious to handle this case.